openvm/tests/apc_builder.rs

```diff
@@ -323,3 +323,176 @@ mod complex_tests {
         assert_machine_output(program.to_vec(), "memcpy_block");
     }
 }
+
+mod pseudo_instruction_tests {
+    use crate::assert_machine_output;
+    use powdr_openvm::symbolic_instruction_builder::*;
+
+    // Arithmetic pseudo instructions
+    #[test]
+    fn mv() {
+        // mv rd, rs1 expands to: addi rd, rs1, 0
+        let program = [
+            // [x8] = [x5]
+            add(8, 5, 0, 0),
+        ];
+        assert_machine_output(program.to_vec(), "mv");
+    }
+
+    #[test]
+    fn not() {
+        // not rd, rs1 expands to: xori rd, rs1, -1
+        // -1 in 24-bit 2's complement is 0xFFFFFF
+        let minus_one: u32 = 0xFFFFFF;
+        let program = [
+            // [x8] = ~[x5]
+            xor(8, 5, minus_one, 0),
+        ];
+        assert_machine_output(program.to_vec(), "not");
+    }
+
+    #[test]
+    fn neg() {
+        // neg rd, rs1 expands to: sub rd, x0, rs1
+        let program = [
+            // [x8] = -[x5]
+            sub(8, 0, 5, 1),
+        ];
+        assert_machine_output(program.to_vec(), "neg");
+    }
+
+    // Set pseudo instructions
+    #[test]
+    fn seqz() {
+        // seqz rd, rs1 expands to: sltiu rd, rs1, 1
```

Comment:
so is it sltiu or is it sltu?

Reply:
Should be the same instruction in OpenVM: The last argument of the `sltu` below is the address space of the third argument. 0 means that it should be an immediate.

Reply:
oh interesting!